softioc/softioc.py

```diff
@@ -233,10 +244,10 @@ def call_f(*args):
 # Hacked up exit object so that when soft IOC framework sends us an exit command
 # we actually exit.
 class Exiter:
-    def __repr__(self):
-        safeEpicsExit()
-    def __call__(self):
-        safeEpicsExit()
+    def __repr__(self):  # hack to exit when "called" with no parenthesis
+        sys.exit(0)
```

Comment:
I guess this will make `code.interact` raise `SystemExit` which will call `safeEpicsExit`, Why do we call sys.exit() here rather than safeEpicsExit?

Reply:
`sys.exit()` will work the same regardless of how `interactive_ioc()` is called.  This keeps the handling of `call_exit=True` vs. `call_exit=False` contained to `interactive_ioc()` where `SystemExit` is now caught.